test_engine.py

```diff
@@ -0,0 +1,372 @@
+from liver_imaging_analysis.engine.dataloader import Keys
+from liver_imaging_analysis.engine.engine import Engine, set_seed
+from monai.transforms import (
+    Compose,
+    EnsureChannelFirstD,
+    ForegroundMaskD,
+    LoadImageD,
+    NormalizeIntensityD,
+    RandFlipd,
+    RandRotated,
+    ResizeD,
+    ToTensorD,
+    RandAdjustContrastd,
+    RandZoomd,
+    ActivationsD,
+    AsDiscreteD,
+)
+#######################################################################################################################
+def set_configs():
+    """
+        Description: Sets new values for config parameters.
+
+        Parameters: None
+
+        Returns: None
+        """
+    config.dataset['prediction'] = "test cases/sample"
+    config.device="cpu"
+    config.dataset['training'] = "Temp2D/Train/"
+    config.dataset['testing'] = "Temp2D/Test/"
+    config.training['batch_size'] = 8
+    config.training['scheduler_parameters'] = {
+                                                "step_size" : 20,
+                                                "gamma" : 0.5, 
+                                                "verbose" : False
+                                                }
+    config.network_parameters['dropout'] = 0
+    config.network_parameters['spatial_dims']= 3 
+    config.network_parameters['channels'] = [8,16,32,64]
+    config.network_parameters['strides'] =  [2, 2, 2]
+    config.network_parameters['num_res_units'] =  0
+    config.network_parameters['norm'] = "INSTANCE"
+    config.network_parameters['bias'] = True
+
+#######################################################################################################################
+@pytest.fixture   
+def engine():
+    """
+        Description: Pytest fixture that initializes an instance of the Engine class with the configured parameters.
+
+        Parameters: None
+
+        Returns: An instance of the Engine class.
+     """
+    set_configs()
+    set_seed()
+    engine= Engine()
+    return engine
+
+#######################################################################################################################
+def test_get_optimizer_adam(engine):
+    """
+    Description: Tests the get_optimizer method of the engine with the Adam optimizer.
+
+    Parameters: engine(object) : An instance of the Engine class.
+
+    Returns: None
+    """
+
+    opt = engine.get_optimizer('Adam', lr=0.001)
+    assert isinstance(opt, torch.optim.Adam)
+    assert opt.defaults['lr'] == 0.001
+
+def test_get_optimizer_sgd(engine):
+    """
+   Description: Tests the get_optimizer method of the engine with the SGD optimizer.
+
+    Parameters: engine(object) : An instance of the Engine class.
+
+    Returns: None
+    """
+    opt = engine.get_optimizer('SGD', lr=0.01, momentum=0.9)
+    assert isinstance(opt, torch.optim.SGD)
+    assert opt.defaults['lr'] == 0.01
+    assert opt.defaults['momentum'] == 0.9
+    
+def test_get_scheduler_steplr(engine):
```

Comment:
This way, you are testing each getter function independently. Maybe you should test their functionality within the engine itself. For example, make sure that upon setting the configs and creating an engine instance, its attributes will carry the expected values returned from the corresponding getters.

Reply:
done